project_mis_report_budget/models/project_project.py

```diff
@@ -48,12 +51,16 @@ def write(self, vals):
             for record in self:
                 real_expenses = 0
                 real_debit_expenses = 0
-                future_expenses = 0
+                expenses = 0
                 hour_expenses = 0
 
                 budget_ids = record.mis_budget_ids.filtered(lambda x: x.state == 'confirmed')
                 incomes = sum(budget_ids.item_ids.filtered(lambda x: x.kpi_expression_id.kpi_id.kpi_type == 'income').mapped('amount'))
-                future_expenses = sum(budget_ids.item_ids.filtered(lambda x: x.date_to > record.last_close_date and x.kpi_expression_id.kpi_id.kpi_type == 'expense').mapped('amount'))
+                expenses = sum(budget_ids.item_ids.filtered(lambda x: x.kpi_expression_id.kpi_id.kpi_type == 'expense').mapped('amount'))
+                past_expenses = budget_ids.item_ids.filtered(lambda x: x.date_to <= record.last_close_date and x.kpi_expression_id.kpi_id.kpi_type == 'expense').mapped('amount')
+
+                for expense in past_expenses:
+                    expenses -= expense

```

Comment:
Why not this (more homogeneous way)?
```suggestion
                past_expenses = sum(budget_ids.item_ids.filtered(lambda x: x.date_to <= record.last_close_date and x.kpi_expression_id.kpi_id.kpi_type == 'expense').mapped('amount'))

                expenses -= past_expenses

```